src/io/csv/type_conversion.cuh

```diff
@@ -0,0 +1,299 @@
+/* Copyright 2017 NVIDIA Corporation.  All rights reserved. */
+
+#ifndef CONVERSION_FUNCTIONS_CUH
+#define CONVERSION_FUNCTIONS_CUH
+
+#include <math_functions.h>
+
+
+__device__
+void removePrePostWhiteSpaces(char *data, long* start_idx, long* end_idx) {
+	while(data[*start_idx] == ' ' && *start_idx < *end_idx)
+		*start_idx=*start_idx+1;
+	while(data[*end_idx] == ' ' && *start_idx < *end_idx)
+		*end_idx=*end_idx-1;
+}
+
+
+template<typename T>
+__device__
+T convertStrtoInt(char *data, long start_idx, long end_idx) {
+
+	T answer = (T)0;
+
+	removePrePostWhiteSpaces(data, &start_idx, &end_idx);
+
+	// if the start and end indexs are the same, then it is a single digit value
+	if (start_idx == end_idx) {
+		answer = (data[start_idx] -'0');
+		return answer;
+	}
+
+	// the data is in little ending, so the last item of data is the lowest digit
+	int powSize = 0;
+	long idx = end_idx;
+
+	if (data[idx] == ' ')
+		--idx;
+
+	while(idx > (start_idx - 1))
+	{
+		if (data[idx] != ',') {
+			answer += (data[idx] -'0') * pow(10, powSize);
+			++powSize;
+		}
+
+		--idx;
+	}
+
+    return answer;
+}
+
+
+template<typename T>
+__device__
+T convertStrtoFloat(char *data, long start_idx, long end_idx) {
+
+	T answer = (T)0.0;
+	removePrePostWhiteSpaces(data, &start_idx, &end_idx);
+
+
+	// check for single digit conversions
+	if (start_idx == end_idx) {
+		answer = (data[start_idx] -'0');
+		return answer;
+	}
+
+	// trim leading and trailing spaces
+	if (data[start_idx] == ' ')
+		++start_idx;
+
+	if (data[end_idx] == ' ')
+		--end_idx;
+
+	// find the decimal point - might not be one
+	long decimal_pt = end_idx;
+	long d_idx = start_idx;
+	int found = 0;
+
+	while ( (d_idx < (end_idx +1)) && ! found  ) {
+		if ( data[d_idx] == '.') {
+			decimal_pt = d_idx;
+			found = 1;
+		}
+		++d_idx;
+	}
+
+	// work on upper part
+	long idx = decimal_pt;
+	int powSize = 0;
+
+	if ( idx != start_idx ) {
+		if (data[idx] == '.')
+			--idx;
+
+		while(idx > (start_idx - 1))
+		{
+			if (data[idx] != ',') {
+				answer += (data[idx] -'0') * pow(10, powSize);
+				++powSize;
+			}
+			--idx;
+		}
+	}
+
+	//lower part - work left to right
+	if ( found ) {
+		powSize = -1;
+		idx = decimal_pt +1;
+
+		while(idx < (end_idx + 1))
+		{
+			if (data[idx] != ',') {
+				answer += (data[idx] -'0') * pow(10, powSize);
+				--powSize;
+			}
+
+			++idx;
+		}
+	}
+
+
+    return answer;
+}
+
+
+/**
+ * Convert a date (MM/YYYY or DD/MM/YYYY) into a date64
+ */
+__device__
+int64_t convertStrtoDate(char *data, long start_idx, long end_idx) {
```

Comment:
This function doesn't seem to have enough precision to capture things like leap seconds or be able to read month names, etc.  I am not sure what is in scope here.

Reply:
limit scope, focused just on two formats - without month names and time portion set to midnight.  This is to address the needs of the demo.
Next version will support full date-time formats